src/components/ModalWithImage.js

```diff
@@ -1,14 +1,20 @@
 import Modal from "./Modal.js";
 
 export default class ModalWithImage extends Modal {
+  constructor(modalSelector) {
+    super({ modalSelector });
+    this._modalImage = this._modalElement.querySelector(".modal__preview");
+  }
   open({ name, link }) {
-    const image = this._modalElement.querySelector(".modal__preview-image");
-    const caption = this._modalElement.querySelector(
+    const imagePreview = this._modalImage.querySelector(
+      ".modal__preview-image",
+    );
+    const caption = this._modalImage.querySelector(
       ".modal__preview-description",
     );
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.